### ISSUE TITLE
Misleading note for error message in proc-macro

### ISSUE THREAD

**Issue by antoyo** [rust-lang/rust#47882]
Hi.
I created a [repository](https://github.com/antoyo/rust-proc-macro-note-bug) that makes Rust show the following error message:
```
error[E0308]: mismatched types
 --> src/main.rs:7:1
  |
7 | #[my_attr]
  | ^^^^^^^^^^
  | |
  | expected &str, found struct `std::string::String`
  | help: consider borrowing here: `&#[my_attr]`
  |
  = note: expected type `&str`
             found type `std::string::String`
```
The help note is misleading and, obviously, does not work.
Thanks to fix this.

**Comment by estebank** [comment]
Duplicate of #48364, needs to change the following to not include the suggestion when dealing with a macro call:
https://github.com/rust-lang/rust/blob/9fd7da904b46ff7aa78c2e2cc1986c4975aeccc6/src/librustc_typeck/check/demand.rs#L259

**Comment by estebank** [comment]
Should have been fixed when #48364 was fixed.
